src/python/pants/option/global_options.py

```diff
@@ -1452,6 +1452,13 @@ def register_options(cls, register):
         advanced=True,
     )
 
+    memory_summary = BoolOption(
+        "--memory-summary",
+        default=False,
+        help=("Report a summary of memory usage at the end of each run."),
+        advanced=True,
+    )
```

Comment:
Hm this naturally fits a lot better in `[stats]`. It looks like we don't pass this option through to the engine, so perhaps it can in fact move? 

Should this be marked `daemon=True` also?

Reply:
Yea, fine with moving.

Shouldn't need `daemon=True`: the value is reported per-run, and doesn't require adjustment to the scheduler.